sources/PTYTextView.m

```diff
@@ -4688,16 +4688,18 @@ - (void)_settingsChanged:(NSNotification *)notification {
 #pragma mark - iTermSelectionDelegate
 
 - (void)selectionDidChange:(iTermSelection *)selection {
+    NSString *selectionString = nil;
     DLog(@"selectionDidChange to %@", selection);
     [_delegate refresh];
     if (!_selection.live && selection.hasSelection) {
         const NSInteger MAX_SELECTION_SIZE = 10 * 1000 * 1000;
-        NSString *selection = [self selectedTextWithCappedAtSize:MAX_SELECTION_SIZE minimumLineNumber:0];
-        if (selection.length == MAX_SELECTION_SIZE) {
-            selection = nil;
+        selectionString = [self selectedTextWithCappedAtSize:MAX_SELECTION_SIZE minimumLineNumber:0];
+        if (selectionString.length == MAX_SELECTION_SIZE) {
+            selectionString = nil;
```

Comment:
The reason `setLastSelection:` takes nil if it’s oversize is to avoid pasting a truncated value later on (the “last selection” is used when pasting from last-selected-text rather than from pasteboard). This isn’t a problem for the variable, though, so let's make sure the value passed to the delegate is never nil.

Reply:
I was purposely leaving the operation of lastSelection alone, assuming something along those lines (with just the variable name changed so not to shadow the method's parameter).

With regards to ensuring the delegate isn't provided nil:  That would be awkward as if the `(!_selection.live && selection.hasSelection)` clause is not true there is no NSString available to pass to the delegate –  we'd need to create a new empty string for the delegate call should that clause be false.

I guess I could do:

```
[_delegate textViewSelectionDidChangeToTruncatedString:selectionString ? selectionString : @""];
```

Which is basically the same as what I have now in the delegate – but I really dislike knowing that the delegate would SEG fault if it were called with nil – perhaps by some future function.

Reply:
If you initialize selectionString to `@""` on line 4691 and then avoid nilling it prior to the call to `setLastSelection:` you can satisfy both the need to pass `nil` to `setLastSelection:` on truncate and the need to always pass the selection to `textViewSelectionDidChangeToTruncatedString:`. Something like
```
NSString *selectionString = @"";
…
  if (!_selection.live && selection.hasSelection) {
    …
    selectionString = [self selectedTextWithCappedAtSize:MAX_SELECTION_SIZE minimumLineNumber:0];
    if (selectionString.length == MAX_SELECTIONSIZE) {
      [[iTermController sharedInstance] setLastSelection:nil];
    } else {
      [[iTermController sharedInstance] setLastSelection:selectionString];
    }
  }
  [_delegate textViewSelectionDidChangeToTruncatedString:selectionString];
```

Reply:
OK, updated as per your suggestion – although I used a ternary operator  on `setLastSelection` to remove duplication.